qemu/tests/nic_acpi_index.py

```diff
@@ -0,0 +1,136 @@
+import logging
+import random
+import re
+
+from avocado.utils import process
+
+from virttest import utils_net
+from virttest import virt_vm
+from virttest import utils_misc
+from virttest import error_context
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    When "acpi-index=N" is enabled, NIC name should always be "ethN"
+
+    1) Boot up guest with a single nic, add nic option "acpi-index=1"
+    2) Remove "biosdevname=0" and "net.ifname=0" from kenrel command line
+    3) Reboot guest
+    4) Check the nic name, the guest nic name enoN == acpi-index=N
+    5) Boot up guest without nic
+    6) Add nic devices through monitor cmd and check if it added
+    7) Check the nic name, the guest nic name enoN == acpi-index=N
+    8) Check if new interface gets ip address
+    9) Ping guest new ip from host
+    10) Delete nic device, then hot plug nic with same slot and different
+    acpi-index number, repeat step 7~9
+    11) Hot plug the nic with different slot and same acpi-index number, repeat
+    step 7~9
+
+    :param test:   QEMU test object.
+    :param params: Dictionary with the test parameters.
+    :param env:    Dictionary with test environment.
+    """
+
+    def renew_ip_address(session, mac):
+        ifname = utils_net.get_linux_ifname(session, mac)
+        make_conf = "nmcli connection add type ethernet con-name %s ifname " \
+                    "%s autoconnect yes" % (ifname, ifname)
+        arp_clean = "arp -n|awk '/^[1-9]/{print \"arp -d \" $1}'|sh"
+        session.cmd_output_safe(make_conf)
+        session.cmd_output_safe("ip link set dev %s up" % ifname)
+        session.cmd_output_safe("dhclient -r", timeout=240)
+        session.cmd_output_safe("dhclient %s" % ifname, timeout=240)
+        session.cmd_output_safe(arp_clean)
+
+    def verified_nic_name():
+        ifname = utils_net.get_linux_ifname(session,
+                                            vm.get_mac_address())
+        pattern = re.findall(r'\d+', ifname)[-1]
+        if pattern == "2":
```

Comment:
It's better to configure in cfg with `acpi-index` together.

Reply:
Changed it with this update.